CONTRIBUTING.md

```diff
@@ -25,13 +26,88 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 ## Getting Started
 
-Dev guides:
-
-- [Contribution Guide]
-- [MacOS Dev Guide]
-- [Windows Dev Guide]
-
-General resources for contributors:
+### Forking Kustomize and Working Locally
+The Kustomize project uses a "Fork and Pull" workflow that is standard to GitHub. In git terms, your personal fork is referred to as the "origin" and the actual project's git repository is called "upstream". To keep your personal branch (origin) up to date with the project (upstream), it must be configured within your local working copy.
+
+### Create a fork in GitHub
+1. Visit https://github.com/kubernetes-sigs/kustomize
+2. Click the `Fork` button on the top right
+
+### Clone the repository
+```bash
+# Clone your repository fork from the previous step
+git clone --recurse-submodules git@github.com:<your github username>/kustomize.git
+cd kustomize
+
+# Configure upstream
+git remote add upstream https://github.com/kubernetes-sigs/kustomize
```

Comment:
Genuine question (you don't have to change anything) - Does it make a difference if we use the ssh URL for the upstream vs the http one? I have it configured with the ssh one, but I'm not sure if that actually affects anything in our workflows

Reply:
The only difference I think I've observed is that
* ssh: you need to authenticate for push and pull, but you can use 2 factor
* http: you only need to authenticate for push, but you need to enter your password
Given that we don't need to push to upstream, I think http is probably easier